Source/WebCore/page/History.cpp

```diff
@@ -195,7 +195,7 @@ ExceptionOr<void> History::go(Document& document, int distance)
 
 URL History::urlForState(const String& urlString)
 {
-    auto* frame = this->frame();
+    RefPtr frame = this->frame();
     if (urlString.isNull())
         return frame->document()->url();
     return frame->document()->completeURL(urlString);
```

Comment:
I think we need to null check document (or assert) here, and protect it before invoking `completeURL`.

Reply:
Good point! I now use protectedDocument. For urlForState I removed reffing of the frame since the (only) caller (stateObjectAdded) already has a ref to the frame.